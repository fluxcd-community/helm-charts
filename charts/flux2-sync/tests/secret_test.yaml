suite: test secret deployment
templates:
  - secret.yaml
tests:
  - it: should be empty if secret.create is false
    asserts:
      - hasDocuments:
          count: 0
    set:
      secret.create: false
  - it: should have kind Secret for k8s >= 1.19
    capabilities:
      majorVersion: 1
      minorVersion: 19
    set:
      secret.create: true
      secret.data:
        testkey: testvalue
        testkey2: testvalue2
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - isAPIVersion:
          of: v1
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
    set:
      secret.create: true
      secret.data:
        testkey: testvalue
        testkey2: testvalue2
