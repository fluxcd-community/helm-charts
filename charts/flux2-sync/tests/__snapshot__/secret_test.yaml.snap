should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      testkey: dGVzdHZhbHVl
      testkey2: dGVzdHZhbHVlMg==
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        helm.sh/chart: flux2-sync-1.0.0
      name: RELEASE-NAME
      namespace: NAMESPACE
    type: Opaque
